src/game.cpp

```diff
@@ -2769,55 +2782,65 @@ void Game::toggleFast()
 
 void Game::toggleNoClip()
 {
-	static const wchar_t *msg[] = { L"noclip disabled", L"noclip enabled" };
 	bool noclip = !g_settings->getBool("noclip");
 	g_settings->set("noclip", bool_to_cstr(noclip));
 
 	runData.statustext_time = 0;
-	m_statustext = msg[noclip];
-
-	if (noclip && !client->checkPrivilege("noclip"))
-		m_statustext += L" (note: no 'noclip' privilege)";
+	if (noclip) {
+		if (client->checkPrivilege("noclip")) {
+			showStatusTextSimple("Noclip mode enabled");
+		} else {
+			showStatusTextSimple("Noclip mode enabled (note: no 'noclip' privilege)");
+		}
+	} else {
+		showStatusTextSimple("Noclip mode disabled");
+	}
 }
 
 void Game::toggleCinematic()
 {
-	static const wchar_t *msg[] = { L"cinematic disabled", L"cinematic enabled" };
 	bool cinematic = !g_settings->getBool("cinematic");
 	g_settings->set("cinematic", bool_to_cstr(cinematic));
 
 	runData.statustext_time = 0;
-	m_statustext = msg[cinematic];
+	if (cinematic)
+		showStatusTextSimple("Cinematic mode enabled");
+	else
+		showStatusTextSimple("Cinematic mode disabled");
 }
 
 // Autoforward by toggling continuous forward.
 void Game::toggleAutoforward()
 {
-	static const wchar_t *msg[] = { L"autoforward disabled", L"autoforward enabled" };
 	bool autoforward_enabled = !g_settings->getBool("continuous_forward");
 	g_settings->set("continuous_forward", bool_to_cstr(autoforward_enabled));
 
 	runData.statustext_time = 0;
-	m_statustext = msg[autoforward_enabled ? 1 : 0];
+	if (autoforward_enabled)
+		showStatusTextSimple("Automatic walking enabled");
```

Comment:
The reason i called it 'autoforward' instead of 'autorun' or 'autowalk' is because it can be used while flying, swimming, in ground fast mode or using a vehicle, so 'walking' is unsuitable. 'Automatic forwards ..' is better, it needs to relate to the control not a specific result of the control.